elephant/signal_processing.py

```diff
@@ -136,6 +137,129 @@ def zscore(signal, inplace=True):
         return result
 
 
+def cross_correlation_function(signal, ch_pairs, env=False, nlags=None):
+
+    """
+    Computes unbiased estimator of the cross-correlation function.
+
+    Calculates the unbiased estimator of the cross-correlation function
+    R(tau) = 1/(N-|k|) R'(tau), where R'(tau) = E[x(t)*y(t+tau)] in a
+    pairwise manner, i.e. signal[ch_pairs[0,0]] vs signal2[ch_pairs[0,1]],
+    signal[ch_pairs[1,0]] vs signal2[ch_pairs[1,1]], and so on (see also
+    Hall & River (2009) Spectral Analysis of Signals, Spectral Element Method
+    in Structural Dynamics, Eq. 2.2.3). The cross-correlation function is
+    obtained by scipy's fftconvolve. Time series in signal are zscored
+    beforehand. Alternatively returns the Hilbert envelope of R(tau), which is
+    useful to determine the correlation length of oscillatory signals.
+
+    Parameters
+    -----------
+    signal : neo.AnalogSignal (nt x nch)
+        Signal with nt number of samples that contains nch LFP channels
+    ch_pairs : list (or array with shape (n,2))
+        list with n channel pairs for which to compute cross-correlation,
+        each element of list must contain 2 channel indices
+    env: bool
+        Return Hilbert envelope of cross-correlation function
+        Default: False
+    nlags: int
+        Defines number of lags for cross-correlation function. Float will be
+        rounded to nearest integer. Number of samples of output is 2*nlags+1.
+        If None, number of samples of output is equal to number of samples of
+        input signal, namely nt
+        Default: None
+
+    Returns
+    -------
+    cross_corr : neo.AnalogSgnal (2*nlag+1 x n)
+        Pairwise cross-correlation functions for channel pairs given by
+        `ch_pairs`. If env=True, the output is the Hilbert envelope of the
+        pairwise cross-correlation function. This is helpful to compute the
+        correlation length for oscillating cross-correlation functions
+
+    Example:
+        dt = 0.02
+        N = 2018
+        f = 0.5
+        t = np.arange(N)*dt
+        x = np.zeros((N,2))
+        x[:,0] = 0.2 * np.sin(2.*np.pi*f*t)
+        x[:,1] = 5.3 * np.cos(2.*np.pi*f*t)
+        # Generate neo.AnalogSignals from x
+        signal = neo.AnalogSignal(x, units='mV', t_start=0.*pq.ms,
+            sampling_rate=1/dt*pq.Hz, dtype=float)
+        rho = elephant.signal_processing.cross_correlation_function(
+            signal, [0,1], nlags=150)
+        env = elephant.signal_processing.cross_correlation_function(
+            signal, [0,1], nlags=150, env=True)
+        plt.plot(rho.times, rho)
+        plt.plot(env.times, env) # should be equal to one
+        plt.show()
+    """
+
+    # Make ch_pairs a 2D array
+    pairs = np.array(ch_pairs)
+    if pairs.ndim == 1:
+        pairs = pairs[:, np.newaxis]
+
+    # Check input
+    if not isinstance(signal, neo.AnalogSignal):
```

Comment:
Do you have a raises section (Raises) in the documentation, you should mention there the `ValueError` and `KeyError`.

Reply:
added raises section